File: C++/ProdusVechi.cpp
#include "ProdusVechi.h"
#include <iostream>
using namespace std;
void ProdusVechi::afisare(float grame)const
{
	cout << "Fiecare produs are o greutate de " << grame << " grame/produs(Polimorfism static)" << endl;
	d.afisareDetalii();
	cout << endl;
}

void ProdusVechi::afiseaza() const
{
	cout << "Produs din clasa <<ProdusVechi>>(Polimorfism dinamic)" << endl;
	d.afisareDetalii();
	cout << endl;
}
void ProdusVechi::afis() const
{
	cout << "Produs de clasa ProdusVechi(Interfata)" << endl;
	cout << endl;
}

File: C#/ProdusVechi.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tema_2_
{
    class ProdusVechi : Produs
    {
        public string DataIndisponibilitate { get; set; }
        public int DurataDisponibilitate { get; set; }
        public void Afisare(float Grame)
        {
            Console.WriteLine($"Produsul are {Grame} grame (Polimorfism static)\n");
            D.AfisareDetalii();
            Console.WriteLine();
        }
        public override void Afiseaza()
        {
            Console.WriteLine("Produs din clasa <<ProdusVechi>> (Polimorfism dinamic)\n");
            D.AfisareDetalii();
            Console.WriteLine();
        }
        public override void Afis()
        {
            Console.WriteLine("Produs din clasa <<ProdusVechi>> (Interfata)\n");
        }

    }
}
